File: CompilationTest/MapTest/results/MapTest.go
package main

import "fmt"

type TestClass struct {
}

func NewTestClass() *TestClass {
    this := new(TestClass)
    return this
}

func (this *TestClass) GetResult() int {
    mapObj := map[string]int{
      "x": 5,
    }
    //let containsX = "x" in mapObj;
    //delete mapObj["x"];
    mapObj["x"] = 3
    return mapObj["x"]
}

func (this *TestClass) TestMethod() {
    fmt.Println(fmt.Sprintf("Result = %v", this.GetResult()))
}

func main() {
    defer func() {
      if r := recover(); r != nil {
          fmt.Print("Exception: ", r)
      }
    }()

    c := NewTestClass()
    c.TestMethod();
}

File: CompilationTest/MapTest/results/MapTest.java
import java.util.HashMap;

class TestClass {
    public Integer getResult() throws Exception
    {
        HashMap<String, Integer> mapObj = new HashMap<String, Integer>();
        mapObj.put("x", 5);
        //let containsX = "x" in mapObj;
        //delete mapObj["x"];
        mapObj.put("x", 3);
        return mapObj.get("x");
    }
    
    public void testMethod() throws Exception
    {
        System.out.println("Result = " + this.getResult());
    }
}

class Program {
    public static void main(String[] args) throws Exception {
        try {
            new TestClass().testMethod();
        } catch (Exception err) {
            System.out.println("Exception: " + err.getMessage());
        }
    }
}